api/src/form_schema/csv_to_jsonschema.py

```diff
@@ -0,0 +1,156 @@
+    """
+    Convert a form definition CSV file to JSON Schema using JsonSchemaBuilder.
+
+    Args:
+        csv_file_path: Path to the CSV file containing form field definitions
+
+    Returns:
+        A dictionary representing the JSON Schema
+    """
+    # Initialize the main schema builder
+    schema_builder = JsonSchemaBuilder()
+
+    # Group fields by section for better organization
+    sections: dict[str, list[FieldInfo]] = {}
+    current_section = None
+
+    reader = csv.DictReader(csv_content.splitlines())
+
+    skipped_field_implementations = ["button"]
+
+    for row in reader:
+        if row.get("Agency FieldName", "") == "" and row.get("Agency Field Name", "") == "":
+            logger.warning(f"Skipping row with empty Agency Field Name: {row['Field ID']}")
+            continue
+
+        if row.get("Field Implementation", "").lower() in skipped_field_implementations:
+            logger.info(f"Skipping field {row['Field ID']} of type {row['Field Type']}")
+            continue
+
+        field_info = FieldInfo.from_dict(row)
+
+        # Skip headers and non-data rows
+        if (
+            not field_info.id
+            or field_info.id == "Field ID"
+            or field_info.type in ["Label", "Button"]
+        ):
+            # Check if this might be a section header
+            if field_info.label and "Header" in field_info.label:
+                current_section = field_info.label.replace(" Header", "")
+                sections[current_section] = []
+            continue
+
+        if current_section:
+            sections.setdefault(current_section, []).append(field_info)
+        else:
+            # If no section is active, add directly to the schema
+            add_field_to_builder(schema_builder, field_info)
+
+    # Now process all sections and add them to the main schema
+    for section_name, fields in sections.items():
+        if not fields:
+            continue
+
+        # Create a section builder with title
+        section_builder = JsonSchemaBuilder(title=section_name)
+
+        # Add each field to the section builder
+        for field_info in fields:
+            add_field_to_builder(section_builder, field_info)
+
+        # Create a sanitized section name for the property key
+        section_key = re.sub(r"[^a-zA-Z0-9]", "_", section_name.lower())
+
+        # Add the section as a sub-object to the main schema
+        schema_builder.add_sub_object(section_key, False, section_builder)
+
+    return schema_builder.build()
+
+
+def add_field_to_builder(builder: JsonSchemaBuilder, field_info: FieldInfo) -> None:
+    """Add a field to the appropriate JsonSchemaBuilder based on its type."""
+    field_id = field_info.id
+    required = field_info.required
+    data_type = field_info.data_type
+    field_type = field_info.type
+    min_value = field_info.min_value
+    max_value = field_info.max_value
+    list_of_values = field_info.list_of_values
```

Comment:
We can follow-up on it, but some enums don't have their actual values in the files we process. For example, the only value for country is `"GENC Standard Ed3.0 Update 11"` which you can see under https://apply07.grants.gov/apply/system/schemas/UniversalCodes-V2.0.xsd in the `CountryCodeDataTypeV3` section (US states are also present).

We'll want to handle swapping those out next.